tests/ga/test_update.py

```diff
@@ -1680,15 +1683,85 @@ def _mock_popen(cmd, *args, **kwargs):
+                    self.assertEqual(filtered_mock_iptable_calls[2], accept_tcp_insert_rule,
+                                     "The third command should add the accept rule")
+
+    def test_it_should_not_set_dns_tcp_iptable_if_drop_unavailable(self):
+
+        with TestOSUtil._mock_iptables() as mock_iptables:
+            with _get_update_handler(test_data=DATA_FILE) as (update_handler, _):
+                with patch.object(osutil, '_enable_firewall', True):
+                    # drop rule is not available
+                    mock_iptables.set_command(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=1)
+
+                    update_handler.run(debug=True)
+
+                    drop_check_command = TestOSUtil._command_to_string(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+
+                    # Filtering the mock iptable command calls with only the once related to this test.
+                    filtered_mock_iptable_calls = [cmd for cmd in mock_iptables.command_calls if cmd in [drop_check_command]]
+
+                    self.assertEqual(len(filtered_mock_iptable_calls), 1, "Incorrect number of calls to iptables: [{0}]".format(mock_iptables.command_calls))
+                    self.assertEqual(filtered_mock_iptable_calls[0], drop_check_command,
+                                     "The first command should check the drop rule")
+
+    def test_it_should_not_set_dns_tcp_iptable_if_drop_and_accept_available(self):
+
+        with TestOSUtil._mock_iptables() as mock_iptables:
+            with _get_update_handler(test_data=DATA_FILE) as (update_handler, _):
+                with patch.object(osutil, '_enable_firewall', True):
+                    # drop rule is available
+                    mock_iptables.set_command(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=0)
+                    # non root tcp iptable rule is available
+                    mock_iptables.set_command(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=0)
+
+                    update_handler.run(debug=True)
+
+                    drop_check_command = TestOSUtil._command_to_string(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+                    accept_tcp_check_rule = TestOSUtil._command_to_string(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+
+                    # Filtering the mock iptable command calls with only the once related to this test.
+                    filtered_mock_iptable_calls = [cmd for cmd in mock_iptables.command_calls if cmd in [drop_check_command, accept_tcp_check_rule]]
```

Comment:
Same comment here, if the accept_tcp_append_rule is called, this filter will ignore it and pass the test irrespective. Please fix this too

Reply:
Same as above. Please share your thoughts

Reply:
same as above